### PR TITLE
feat: Add support to check and create storage accounts when accountOptions has accountName

### PR DESCRIPTION
**Pull request by mukhoakash**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by mukhoakash** [comment]
@andyzhangx  I will add the changes to pass contex once [this](https://github.com/kubernetes-sigs/cloud-provider-azure/pull/784) PR gets merged.

Working on adding unittests for the same.

File: pkg/provider/azure_storageaccount.go [hunk 1]
```diff
@@ -118,25 +118,54 @@ func (az *Cloud) GetStorageAccesskey(account, resourceGroup string) (string, err
...
+	if accountOptions == nil {
+		return "", fmt.Errorf("account options is nil")
+	}
+
+	accountName := accountOptions.Name
+
+	if len(accountName) == 0 {
+		return "", fmt.Errorf("Account name not provided in account options")
+	} else {
+		accountExists, err := az.checkIfAccountExists(accountOptions)
```
Comment (andyzhangx) [inline pkg/provider/azure_storageaccount.go:133]: if you only need `checkIfAccountNotExists`, I think you could add this function into your CSI driver, it's not necessary to add this function here.

Comment (mukhoakash) [reply pkg/provider/azure_storageaccount.go:133]: I'll move the function `checkIfAccountNotExists` to the driver. But I would need the function to create the storage account.

Comment (andyzhangx) [reply pkg/provider/azure_storageaccount.go:133]: what about add a new field "CreateIfNotExists"(by default: false) in AccountOptions, that would be a minor change

Comment (mukhoakash) [reply pkg/provider/azure_storageaccount.go:133]: The EnsureStorageAccount function is designed in a way to work when accountName is empty in accountOptions. For my case, I would need to create an account with that particular accountName if it doesnt exist. So, the `createStorageAccount` function might be needed after all?

**Comment by mukhoakash** [comment]
Abandoning this PR since we decided on modifying the EnsureStorageAccount function to accomodate the requirement.

**Pull request closed without merging**